Clarify usage in Readme for 0.10.x users

alfwatt opened the pull request.

Commit:
Update readme to clarify usage

Commit:
Reflow for ease of git diff reading

rclee commented on readme.md:
```diff
@@ -34,41 +38,67 @@ let eventsManager = MMEEventsManager.sharedManager().initialize(
 
-The MapboxMobileEvents frameworks collect location data to help us improve the map. We strive to maintain a low power and network usage profile for this collection and take great care to anonymize all data in accordance with our [privacy policy](https://www.mapbox.com/legal/privacy).
+The MapboxMobileEvents frameworks collect location data to help us improve the map. We strive to maintain a 
+low power and network  usage profile for this collection and take great care to anonymize all data in accordance 
```
```suggestion
low power and network usage profile for this collection and take great care to anonymize all data in accordance 
```

rclee commented on readme.md:
````diff
@@ -34,41 +38,67 @@ let eventsManager = MMEEventsManager.sharedManager().initialize(
+Applications and frameworks which embed `MapboxMobileEvents.framework` can implement the 
+ `MMEEventsManagerDelegate` method:
 
 ```objc
````
Should we mention setting the delegate somewhere around here since that step is removed above?

rclee reviewed (approved):
LGTM. Added some typo fixes and one comment about the delegate. Thanks for keeping the docs shiny @alfwatt!

dmiluski committed:
Extension Example: Today Widget

Goal:
Show that MapboxMobileEvents SDK is able to be built/run in an extension.

Approach:
Provide a Today Widget example loading MapboxMobileEvents SDK and displaying (demonstrated on iPhone SE)

Outcome:
Able to build/run without crashing, although there is a warning due to the use of UIApplication.shared application directly inside of the framework.

> d: warning: linking against a dylib which is not safe for use in application extensions: /Users/danemiluski/Library/Developer/Xcode/DerivedData/MapboxMobileEvents-gtwngrjytwknvkfqciqcsoxcqjvr/Build/Products/Debug-iphonesimulator/MapboxMobileEvents.framework/MapboxMobileEvents

This warning could be remedied by enabling Allow app extension API only. Unfortunately, this would alos prevent the use of UIApplication.shared which can prove useful to identify UIApplication’s state.

dmiluski committed:
iOS (App Extension) Freindliness Continued

Context:
https://github.com/mapbox/mapbox-events-ios/pull/205

Although the above PR allowed for the content scale values to be extracted while in an extension, it did not cover all cases.

MMEUIApplicationWrapper also leverages UIApplication.shared to example application state as well as handle background task communication.

Although this is unavailable in NSExtensions, this can be stubbed out for the time being until a longer term decision is made upon how to handle lengthy actions/collections in an extenion.

For further support:
https://developer.apple.com/library/archive/documentation/General/Conceptual/ExtensibilityPG/ExtensionScenarios.html#//apple_ref/doc/uid/TP40014214-CH21-SW1

dmiluski committed:
[App Extensions] Allow App Extension API Only

Problem:
When running, if marked extension API only, there are several errors to compile when accessing UIApplication.shared application.

This is due to:
- Lack of background task support through shared application
- Lack of app state

Goal:
Provide SDK friendly way to extract app state from sharedApplication is available. Otherwise, fallback to a default value.

Disable background task kick offs for extensions for the time being.

Changes:
- Remove MMEUINavigation at it is not used
+ Mark Allow app extension API only (Will error if build if sharedApplication is used directly providing feedback)
+ Extend MMEUIApplicationWrapper to infer sharedApplication or extension behaviors based on extension or not

dmiluski committed:
[User Defaults Sharing] Sharing UserDefaults between App/Extension

Goal:
Experiment with UserDefault sharing between App/Extension

Outcome:
Through experimentation, confirmed what I wanted to clarify with the team on behaviors.

I access a shared user default between App/Extensions a UserDefaults(suite:) needs to be instantiated with the shared AppGroup provided from the top level application which has been configured with App Group.

If the goal is to support sharing user default values between App/Extension we would need to request our developers set the shared app group and we leverage it to access the shared resource.

Experiment:
Setup App Group and hop between App/Extension reading/writing to see what is visible between the two.

Manual Tests:
Evaluated the behavior of the App and extension by reading and writing. Launching each one after the other to verify whether the other could see the data written by the other

- Mismatch suite to App Group (No data sharing)
- Matchin suite with AppGroup (Data can be read/write between the two)

dmiluski committed:
[Flakey Test] Comment out

Test is refactored in 1.0 branch which should provide non-flakey coverage

dmiluski committed:
[Extension Support] Default Telemetry Off

Goal:
Currently, there is a devveloper broken with their attempt at integrating Maps/Telemetry support with their App extension.

We must also consider the ramifications of supporting telemetry gathering in an extension in general as questions around shared data store of the AppGroup as well as Privacy/Consent values have not been fully worked through yet.

Approach:
To unblock sooner rather than later, offer a supporting solution with an SDK using App Extension Only apis, but disable telemetry gathering if in an extension until interfaces and privacy considered expectations are finalized.

Changes:
- Disable Telemetry Gathering in App Extensions for the time being

dmiluski committed:
PR Feedback:

- Remove commented out code
- Remove commented out/unused property
- Remove unneeded comments
- Remove header doc

Commit:
Circleci project setup (#215)

* Add .circleci/config.yml
* Add the ci-run target to the makefile, see if /script/package.sh -s is sufficient for a CI run
* A more modern Xcode
* Newlines are Free!

rclee committed:
Cherrypick - Reformed UA Fix (#213)

* Check for bundle Ids as well, better filtering and sorting, for more completely consistently reformed user agent strings

* Fix known null issue with kCFBundleNameKey

Co-authored-by: Alf Watt <alf.watt@mapbox.com>

Commit:
Remove MMENamespacedDependencies.h (#216)

Co-authored-by: Randall Lee <randall.lee@mapbox.com>

Commit:
Exposes Private Headers via a Development Target to both Swift and Obj-C test apps (#217)

* A beast of a fix: https://github.com/mapbox/mobile-telemetry/issues/666
* Enable MME_DEPRECATION_WARNINGS on the (Development) target

* Don't set the Run target of the MapboxMobileEvents scheme to MMETest, removing an implicit decency
* Resolve the dependency issues preventing the bitrise build for passing
* Remove empty ViewController class
* TodayWidget deployment target set to the dawn of widgets

Commit:
Bring back setDebugLoggingEnabled, improve category method names (#218)

* Put setDebugLoggingEnabled back
* Safer private method name on this public class category

Commit:
Update readme.md

Co-authored-by: Randall Lee <randall.lee@mapbox.com>

Commit:
Update readme.md

Co-authored-by: Randall Lee <randall.lee@mapbox.com>

Commit:
Update readme.md

Co-authored-by: Randall Lee <randall.lee@mapbox.com>

Commit:
Suggested edit
